src/components/Sidebar/SidebarTile.component.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from 'react';
+
+import { useLocation, useNavigate } from 'react-router';
+
+import {
+    ExpandLess as ExpandLessIcon,
+    ExpandMore as ExpandMoreIcon,
+} from '@mui/icons-material';
+import {
+    Collapse,
+    List,
+    ListItemButton,
+    ListItemIcon,
+    ListItemText,
+} from '@mui/material';
+
+import { StyledErrorBadge } from '@components';
+
+import { type SidebarTileProps } from './Sidebar.types';
+
+/**
+ * Sidebar Tile component
+ * Provides List Item tile with icon, label, onClick, and collapse
+ * @returns Complete Sidebar List item customized as per props
+ *
+ * @example usage
+ * tsx```
+ * <SidebarTile {...sidebarTileItem} onclick={handleClick}/>
+ * ```
+ */
+export const SidebarTile = (props: SidebarTileProps) => {
+    const [isCollapseOpen, setCollapseOpen] = useState<boolean>(false);
+    const navigate = useNavigate();
+    const { pathname } = useLocation();
+    if (props.purpose === 'divider') {
+        return;
+    }
+    const { icon: PropsIcon, label, route, badge, children, onClick } = props;
+    const isActive =
+        route === pathname ||
+        children?.some((child) => child.route === pathname);
+    const handleCollapseClick = () => {
```

Comment:
Add newline on line number 46 and just after handleCollapseClick function and also always before return (for return check if there is an eslint rule for that with we can achieve the same).

```suggestion
        children?.some((child) => child.route === pathname);

    const handleCollapseClick = () => {
```